curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/understanding-the-differences-between-the-freecodecamp-and-browser-console.english.md

```diff
@@ -9,15 +9,15 @@ forumTopicId: 301193
 <section id='description'>
 You may have noticed that some freeCodeCamp JavaScript challenges include their own console. This console behaves a little differently than the browser console you used in the last challenge.
 The following challenge is meant to highlight the main difference between the freeCodeCamp console and your browser console.
-When you run ordinary JavaScript, the browsers console will display your <code>console.log()</code> statements the exact number of times you requested.
-The freeCodeCamp console will print your <code>console.log()</code> statements for each test of that challenge, as well as one more time for any function calls that you have in your code.
-This lends itself to some interesting behavior and might trip you up in the beginning, because a logged value that you expect to see only once may print out many more times.
-If you would like to see only your single output and not have to worry about running through the test cycles, you can use <code>console.clear()</code> and check the browsers console.
+When you run ordinary JavaScript, the browser's console will display your <code>console.log()</code> statements the exact number of times it is called.
+The freeCodeCamp console will print your <code>console.log()</code> statements a short time after the editor detects a change in the script, as well as during every test.
+The freeCodeCamp console is cleared before every test. So, you may miss a log, if the tests finish too quickly.
```

Comment:
```suggestion
The freeCodeCamp console is cleared before the tests are run and, to avoid spam, only prints the logs during the first test. The exceptions to this are logs called inside functions, which are printed whenever the function is called.
```
The second sentence might be better as a footnote, since it takes you out of the flow. Thoughts?  Nonetheless logs inside functions are always shown *if* those functions are executed during tests.  The idea being users might want to know what their functions are doing when tested, so we should not hide that.

Reply:
I would be happy with it either as you have suggested, or as a footnote. By "footnote" do you mean a final paragraph with the beginning: **Note:**  ?

Reply:
Initially I thought a literal footnote, like <sup>1</sup>: \<footnote context\>, but I don't think that's used in any other challenges so I think a **Note:** is a better bet. I've edited that in.